### PR TITLE
Add Role Whitelisting

### PR DESCRIPTION
**Pull request by ShadowJonathan**
This PR adds:
- `pls whitelist <purge/add/remove> [role]`
- An extra array field per guildEntry
- A check for the whitelist
- Gayness

Essentially if the whitelist array has more than 0 objects in it, it will be enabled, and then only everyone either having that role, or `administrator`, will be allowed to send commands.

### CODE REVIEW

File: src/commands/configCommands/whitelist.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          return `No role found with name '${name}', check if you got punctuation alright, it's not case-sensitive.`;
+        }
+      }
+    }
+
+    let response;
+
+    switch (args[0]) {
+      case 'add':
+        guildEntry.addWhitelistedRole(role.id);
```
Comment (ParadoxOrigins) [inline src/commands/configCommands/whitelist.js]:
```suggestion
        await guildEntry.addWhitelistedRole(role.id).save()
```
You can chain this, it saves a few lines

**Comment by ShadowJonathan** [comment]
Got it, let me restructure some stuff then

**Comment by melmsie** [comment]
Seems fine enough for me, will probably tweak wording later though.

**Comment by ShadowJonathan** [comment]
Tried to add the flavour or "meme-ness" into the descriptions and such, but that's alright.

**Pull request merged**